File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OnOfflineTilesRemovedCallback.java
package com.mapbox.services.android.navigation.v5.navigation;

import com.mapbox.geojson.BoundingBox;

/**
 * Listener that needs to be added to
 * {@link MapboxOfflineRouter#removeTiles(String, BoundingBox, OnOfflineTilesRemovedCallback)} to know when the routing
 * tiles within the provided {@link BoundingBox} have been removed
 */
public interface OnOfflineTilesRemovedCallback {

  /**
   * Called when the routing tiles within the provided {@link BoundingBox} have been removed completely.
   *
   * @param numberOfTiles removed within the {@link BoundingBox} provided
   */
  void onRemoved(long numberOfTiles);
}

File: libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/OnOfflineTilesRemovedCallback.kt
package com.mapbox.navigation.route.onboard

import com.mapbox.geojson.BoundingBox

/**
 * Listener that needs to be added to
 * [com.mapbox.navigation.route.onboard.task.RemoveTilesTask] to know when the routing
 * tiles within the provided [BoundingBox] have been removed
 */
internal interface OnOfflineTilesRemovedCallback {

    /**
     * Called when the routing tiles within the provided [BoundingBox] have been removed completely.
     *
     * @param numberOfTiles removed within the [BoundingBox] provided
     */
    fun onRemoved(numberOfTiles: Long)
}
